config/credhub_secret_test.go

```diff
@@ -82,7 +82,75 @@ var _ = Describe("CredhubSecret", func() {
 
 			BeforeEach(func() {
 				fakeAuthClient.DoAuthenticatedPutReturns(403, e)
-				secretsError = credhubSecret.Add("old-service-name", "application/cloud/one", "{\"key\":\"secret\"}")
+				secretsError = credhubSecret.Add("old-service-name", "application/cloud/master/one", "{\"key\":\"secret\"}")
+			})
+
+			It("should return error message", func() {
+				Expect(secretsError).To(Equal(e))
+			})
+		})
+
+		Context("when add calls with invalid path with less than three elements", func() {
+
+			var e = errors.New("CredHub path does not include required fields: {appName}/{profile}/{label}/{propertyName}")
+
+			JustBeforeEach(func() {
+				secretsError = credhubSecret.Add("old-service-name", "application/default/one", "{\"key\":\"secret\"}")
+			})
+
+			BeforeEach(func() {
+				fakeAuthClient.DoAuthenticatedPutReturns(200, nil)
+			})
+
+			It("should return error message", func() {
+				Expect(secretsError).To(Equal(e))
+			})
+		})
+
+		Context("when add calls with invalid path without elements", func() {
+
+			var e = errors.New("CredHub path does not include required fields: {appName}/{profile}/{label}/{propertyName}")
+
+			JustBeforeEach(func() {
+				secretsError = credhubSecret.Add("old-service-name", "///", "{\"key\":\"secret\"}")
+			})
+
+			BeforeEach(func() {
+				fakeAuthClient.DoAuthenticatedPutReturns(200, nil)
+			})
+
+			It("should return error message", func() {
+				Expect(secretsError).To(Equal(e))
+			})
+		})
+
+		Context("when add calls with invalid path with more than three elements", func() {
+
+			var e = errors.New("CredHub path does not include required fields: {appName}/{profile}/{label}/{propertyName}")
```

Comment:
This error message is confusing if the user has specified more than the expected number of elements. The path _does_ include the required fields - it just includes other ones too 😀

Reply:
CredHub path does not include required fields -> CredHub path should just include the required fields?